cloudinit/sources/helpers/azure.py

```diff
@@ -48,6 +49,9 @@
     description="initialize reporter for azure ds",
     reporting_enabled=True)
 
+DEFAULT_REPORT_FAILURE_USER_VISIBLE_MESSAGE = (
+    'The VM encountered an error during provisioning.')
```

Comment:
@anhvoms @OddBloke Pending Azure team discussions on what this customer visible message should be.

Reply:
Done. The user visible error message will be `The VM encountered an error during deployment. Please visit https://aka.ms/linuxprovisioningerror for more information on remediation.`